### DISCUSSION TITLE
[Bug] Azure Openai 部署后，只有 3.5 可以用

### DISCUSSION THREAD

**Discussion by itcodes** [lobehub/lobehub#1674] [Self Hosting] [answered]
### 💻 系统环境

Other

### 📦 部署环境

Vercel / Zeabur / Sealos

### 🌐 浏览器

Chrome

### 🐛 问题描述

![image](https://github.com/lobehub/lobe-chat/assets/8282645/1a0a157b-0ef4-44f6-962d-445c943dc593)
部署后，选择 3.5 是可以用，但是其他 2 个会报错。
问题是 Azure 后台部署后确实是这样？这种怎么解决？
感谢各位大佬。
![image](https://github.com/lobehub/lobe-chat/assets/8282645/c57943fc-1743-4661-8cdf-09eb79d33c83)

[a comment by an automated account is not shown]

**Comment by itcodes** [comment]
LobeChat 约定了与模型同名的部署名才能正常使用。那么类似gpt-4-0125-preview，在 Azure 中他就是 gpt4，只不过模型版本是 0125，如何调整呀？感谢。

[a comment by an automated account is not shown]

**Comment by mingyue0094** [comment]
设置 - 通用模型 -自定义模型名称 ， 添加  Azure 后台显示的模型名称， 然后就可以选了。 另外好像 docker 启动时，可以加参数添加自定义模型。

[a comment by an automated account is not shown]

**Comment by itcodes** [comment]
> 设置 - 通用模型 -自定义模型名称 ， 添加 Azure 后台显示的模型名称， 然后就可以选了。 另外好像 docker 启动时，可以加参数添加自定义模型。

没看到你说的这个自定义模型在哪里呀？我在部署的时候就配置了。

[a comment by an automated account is not shown]

**Comment by fooooxxxx** [comment, marked as answer]
去 azure 新建部署,部署名填写 `gpt-4-turbo-preview`, 模型选择`gpt-4`,模型版本选择`0613 默认`, 然后点击创建

要用 gpt-4-vision-preview 的话, 再次在 azure 中新建部署, 部署名填写 `gpt-4-vision-preview`，模型选择`gpt-4`, 模型版本选择`vision-preview`, 点击创建

https://chat-docs.lobehub.com/zh/self-hosting/examples/azure-openai 原因参考这个

**Comment by itcodes** [comment]
> 去 azure 新建部署,部署名填写 `gpt-4-turbo-preview`, 模型选择`gpt-4`,模型版本选择`0613 默认`, 然后点击创建
> 
> 要用 gpt-4-vision-preview 的话, 再次在 azure 中新建部署, 部署名填写 `gpt-4-vision-preview`，模型选择`gpt-4`, 模型版本选择`vision-preview`, 点击创建
> 
> https://chat-docs.lobehub.com/zh/self-hosting/examples/azure-openai 原因参考这个

感谢。已解决。这跟 nextweb 还不太一样。另外想问下。如果用了CUSTOM_MODELS，-all,+gpt-4-turbo-preview后，为啥默认的 版本还是 gpt-3.5-turbo

**Comment by fooooxxxx** [comment]
> 感谢。已解决。这跟 nextweb 还不太一样。另外想问下。如果用了CUSTOM_MODELS，-all,+gpt-4-turbo-preview后，为啥默认的 版本还是 gpt-3.5-turbo

这个我没关注过。你可以试试环境变量里使用`DEFAULT_AGENT_CONFIG` 配置`model`为`gpt-4-turbo-preview`
 https://chat-docs.lobehub.com/zh/self-hosting/environment-variables/basic#default_agent_config

**Comment by CoreJa** [comment]
实际上现在版本依然有这个问题，DEFAULT_AGENT_CONFIG设置model后，它选择的model似乎还是openAI的model，而非我配置的azure model。然后会问我要openapi key:

![image](https://github.com/user-attachments/assets/d1e73d8d-58f1-4526-81ff-1de46c507299)


具体信息：
- AZURE_MODEL_LIST: azure-gpt-4o->gpt-4o=GPT-4o<128000:vision:fc:file>
- DEFAULT_AGENT_CONFIG: model=azure-gpt-4o


另外DEFAULT_AGENT_CONFIG还试过model=gpt-4o，结果同上，都不好使。

简单结论：纯Azure OpenAI无法开箱即用，需要在对话框上面**手动选择**一下azure的model，才能正常使用。

BTW，请问系统助手能配置吗？目前没看到相关文档。

**Reply by arvinxx** [reply]
`DEFAULT_AGENT_CONFIG: model=azure-gpt-4o, provider=azure`

**Reply by CoreJa** [reply]
感谢，应该是分号(`;`)btw(否则页面载不进去），设置后worked well! 另想知道DEFAULT_AGENT_CONFIG只看到了[配置规则](https://lobehub.com/zh/docs/self-hosting/environment-variables/basic#default-agent-config)，但相关配置项在文档里体现的不太充分，请问在哪里可以看到更为详细的配置项呢？thx!
